library/src/main/scala/scala/scalajs/js/JSConverters.scala

```diff
@@ -41,6 +42,46 @@ object JSConverters extends JSConvertersLowPrioImplicits {
     @inline final def toJSArray: Array[T] = genTraversableOnce2jsArray(col)
   }
 
+  implicit class JSRichGenIterable[T](val col: GenIterable[T]) extends AnyVal {
+    @inline final def toJSIterable: Iterable[T] = new IterableAdapter(col)
+  }
+
+  implicit class JSRichIterator[T](
+      val it: scala.collection.Iterator[T]) extends AnyVal {
+    @inline final def toJSIterator: Iterator[T] = new IteratorAdapter(it)
+  }
+
+  @inline
```

Comment:
`@inline` on a JS class doesn't do anything. I suggest you remove it so that we do not pretend it does.